sdk/how-to-guides/getting-started-with-testing.md

```diff
@@ -244,89 +190,79 @@ Things you typically want to test with integration tests:
+
+Once you have used `ops_test` to get a model in which to run your integration tests, most of the remaining integration test code will interact with Juju via the `python-libjuju` package.
+
+> See more: [`python-libjuju`](https://pythonlibjuju.readthedocs.io/en/latest/)
 
 [note]
-If you run `tox -e integration --model foo`, `ops_test` will not generate a random new model, but instead will connect to an existing model called `foo` and the tests will be run there. This means that multiple modules will all be executed in the same model, which can give headaches in CIs, where the `--model` argument is often used to provide a predictable model name for logging collection purposes. This is the case for example in `template-operator`.
+*Pro tip*: you can prevent `ops_test` from tearing down the model on exit by passing the `--keep-models` argument. This is useful when the tests fail and the logs don't provide a sufficient post-mortem and a real live autopsy is required.
 [/note]
 
-> *Pro tip*: you can prevent `ops_test` from tearing down the model on exit by passing the `--keep-models` argument. This is useful when the tests fail and the logs don't provide a sufficient post-mortem and a real live autopsy is required.
-
-Detailed documentation of how to use `ops_test` and `pytest-operator` is out of scope for this document. However, this is how a typical integration test could look like (yes, `pytest-operator` has an async API):
+Detailed documentation of how to use `ops_test` and `pytest-operator` is out of scope for this document. However, this is an example of a typical integration test:
 
 ```python
 async def test_operation(ops_test: OpsTest):
-    # tweak the config
+    # Tweak the config:
     app: Application = ops_test.model.applications.get("tester")
     await app.set_config({"my-key": "my-value"})
     
-    # add another charm and relate them
+    # Add another charm and relate them:
     await ops_test.model.deploy('other-app')
     await ops_test.model.relate('tester:endpoint1', 'other-charm:endpoint2')
     
-    # scale it up
+    # Scale it up:
     await app.add_unit(2)
     
-    # remove another app
+    # Remove another app:
     await ops_test.model.remove_application('yet-another-app')
     
-    # run an action on a unit
+    # Run an action on a unit:
     unit: Unit = app.units[1]
     action = await unit.run('my-action')
     assert action.results == <foo>
     
-    # what this means depends on the workload
+    # What this means depends on the workload:
     assert charm_operates_correctly()  
 ```
 
-`ops_test` has, of course, API for all inverse operations: remove an app, scale it down, remove a relation...
+`python-libjuju` has, of course, an API for all inverse operations: remove an app, scale it down, remove a relation...
 
 A good integration testing suite will check that the charm continues to operate as expected whenever possible, by combining these simple elements.
 
-
 # Functional Testing
```

Comment:
This section is a bit odd, because it's linking out to Jon's parca-operator repo as an example of functional testing, but then says to do Scenario tests, and those aren't the same. Jon's functional tests are testing the Parca lib, don't use Harness or Scenario, and I don't think you would rewrite them to do so (no Juju events are involved).

"unit", "functional", "integration" and other test categories are somewhat loosely defined in the industry, which makes this kind of thing tricky.

I think this should be left for the refactor/rethinking, and content like the [Best Practices](https://discourse.charmhub.io/t/best-practices-for-testing-charms/15172) guide could maybe replace this? For testing charms, our current recommendation is to do unit tests (today with both Harness and Scenario, but any-moment-now-(tm) just Scenario) and integration tests (today, with pytest-operator and pylibjuju). If you're testing a charm lib, that probably deserves its own guide, and it depends a lot on whether it's isolated from Juju (like Jon's one) or emits custom events.